manifests/unit_file.pp

```diff
@@ -90,6 +90,14 @@
 
   assert_type(Systemd::Unit, $name)
 
+  if $hasrestart {
```

Comment:
Since it's a boolean, shouldn't this be check if it's not undef:
```suggestion
  if $hasrestart =~ NotUndef {
```

Reply:
ah yes